catalogue/webapp/utils/search.ts

```diff
@@ -1,43 +1,47 @@
-import {
-  CatalogueApiError,
-  CatalogueResultsList,
-  Image,
-  Work,
-} from '@weco/common/model/catalogue';
-import { Story } from '@weco/catalogue/services/prismic/types/story';
-import {
-  PrismicApiError,
-  PrismicResultsList,
-} from '@weco/catalogue/services/prismic/types';
+import { CatalogueApiError } from '@weco/common/model/catalogue';
+import { PrismicApiError } from '@weco/catalogue/services/prismic/types';
 
 export type DefaultSortValuesType = {
   sort: string | undefined;
   sortOrder: string | undefined;
 };
 
-type QueryResultsProps = (Work | Image | Story)[] | undefined;
+type HasResultsList<T> = {
+  results: T[];
+  type: 'ResultList';
+};
 
 /**
  * Takes query result and checks for errors to log before returning required data.
  * @param {string} categoryName - e.g. works
  * @param queryResults - Original result from query
  */
-export const getQueryResults = (
-  categoryName: string,
-  queryResults:
-    | CatalogueResultsList<Image | Work>
-    | CatalogueApiError
-    | PrismicResultsList<Story>
-    | PrismicApiError
-): QueryResultsProps => {
+export function getQueryResults<T>({
+  categoryName,
+  queryResults,
+}: {
+  categoryName: string;
+  queryResults: HasResultsList<T> | CatalogueApiError | PrismicApiError;
+}): T[] | undefined {
```

Comment:
Nice types here!

My very minor nit is that `HasResultsList` should just be `ResultsList`, because it doesn't have a ResultsList, it _is_ a ResultList.

Also you could make a generic error type here, but it's less important.

Reply:
Thank you thank you! Took up your examples. And you're right, I'll change that. 

Happy to make a generic error type, do you mean with the same "model" as I did the general result list?

Reply:
I went with https://github.com/wellcomecollection/wellcomecollection.org/pull/9056/commits/74cb9ed7ac0f7a9a4ce55f84af899bc3d8a24a80#diff-ecee3ed998bd8959458ec3c1d1a8fadd31bc1bb297966eb4b7640ab6a5f9a3e2R11